ui/src/app/components/interactive.tsx

```diff
@@ -0,0 +1,100 @@
+import * as Utils from 'app/utils';
+import * as fp from 'lodash/fp';
+import * as React from 'react';
+
+const {useState, forwardRef} = React;
+const allowedHoverVariables = ['backgroundColor', 'border', 'color', 'boxShadow', 'opacity', 'textDecoration'];
+const pointerTags = ['button', 'area', 'a', 'select'];
+const pointerTypes = ['radio', 'checkbox', 'submit', 'button'];
+
+interface InteractiveProps {
+  className?: string;
+  type?: string;
+  role?: string;
+  onClick?: () => any;
+  onKeyDown?: () => any;
+  onMouseDown?: Function;
+  onBlur?: Function;
+  disabled?: boolean;
+  children?: React.ReactElement | React.ReactElement[];
+  tabIndex?: number;
+  hover?: React.CSSProperties;
+  style?: React.CSSProperties;
+  tagName: keyof JSX.IntrinsicElements;
+}
+
+export const Interactive: React.ForwardRefExoticComponent<InteractiveProps> = forwardRef(({
+  className = '',
+  tagName: TagName = 'div',
+  type,
+  role,
+  onClick,
+  onKeyDown,
+  onMouseDown,
+  onBlur,
+  disabled,
+  children,
+  tabIndex,
+  hover = {},
+  style = {},
+  ...props}: InteractiveProps, ref) => {
+  const [outline, setOutline] = useState(undefined);
+  const { cursor } = style;
+
+  const computedCursor = Utils.cond(
+    [!!cursor, () => cursor],
+    [disabled, () => undefined],
+    [!!onClick || pointerTags.includes(TagName) || pointerTypes.includes(type), () => 'pointer']
+  );
+
+  const computedTabIndex = Utils.cond(
+    [fp.isNumber(tabIndex), () => tabIndex],
+    [disabled, () => -1],
+    [!!onClick, () => 0],
+    () => undefined);
+
+  const computedRole = Utils.cond(
```

Comment:
how is 'role' used?

Reply:
This is the aria role - being clickable it will default to a button if it is one of the tags defined as such. It will honor a manually set role, or if it cannot make a good determination, it will not set the role at all. Basically, this is a best effort to be accessible.

Reply:
nice.  thanks for adding this.